torch/utils/checkpoint.py

```diff
@@ -35,62 +35,84 @@ def check_backward_validity(inputs: Iterable[Any]) -> None:
         warnings.warn("None of the inputs have requires_grad=True. Gradients will be None")
 
 
+def _get_device_module(device='cuda'):
+    device_module = getattr(torch, device)
+    return device_module
+
+
 # We can't know if the run_fn will internally move some args to different devices,
 # which would require logic to preserve rng states for those devices as well.
 # We could paranoically stash and restore ALL the rng states for all visible devices,
 # but that seems very wasteful for most cases.  Compromise:  Stash the RNG state for
 # the device of all Tensor args.
```

Comment:
It appears that in this comment we already do this tradeoff, so I think its okay to make the same trade off here. 

We should just have a note on the checkpoint docs that device state is only preserved for devices of the Tensor args, the workaround if there are no Tensor args is just to explicitly pass in a dummy tensor on the correct device.

Reply:
> It appears that in this comment we already do this tradeoff, so I think its okay to make the same trade off here.
> 
> We should just have a note on the checkpoint docs that device state is only preserved for devices of the Tensor args, the workaround if there are no Tensor args is just to explicitly pass in a dummy tensor on the correct device.

yeah, thanks for your reminder, I hadn't noticed the note. And I made some modifications to extract the device information from the input parameters. @soulitzer